lib/bosh/BoshDirectorClient.js

```diff
@@ -133,45 +249,51 @@ class BoshDirectorClient extends HttpClient {
 
   createOrUpdateDeployment(manifest, opts) {
     const query = opts ? _.pick(opts, 'recreate', 'skip_drain', 'context') : undefined;
+    const deploymentName = yaml.safeLoad(manifest).name;
     return this
-      .request({
+      .makeRequestWithConfig({
         method: 'POST',
         url: '/deployments',
         headers: {
           'Content-Type': 'text/yaml'
         },
         qs: query,
         body: _.isObject(manifest) ? yaml.safeDump(manifest) : manifest
-      }, 302)
-      .then(res => parseInt(this.lastSegment(res.headers.location)));
+      }, 302, this.config)
+      .tap(() => {
+        logger.info(`Cached ${deploymentName} at director: ${this.config.name} ${this.config.url}`);
+        this.cache[deploymentName] = this.config;
```

Comment:
based on the `_.sample(this.activePrimary)` returned in the beginning of method update cache accordingly in the above.